File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    public static double ProductionRatePerHour(int speed)
    {
        var ProductionRatePerHour= 221;
        var scsRate = 0.0;

        if (speed <= 4){
            return (speed*ProductionRatePerHour);
        }
        else if (speed <= 8){
            scsRate = 0.9;
            return (speed * scsRate * ProductionRatePerHour);
        }
        else if (speed <= 9)
        {
            scsRate = 0.8;
            return (speed * scsRate * ProductionRatePerHour);
        }
        else if (speed <= 10)
        {
            scsRate = 0.77;
            return (speed * scsRate * ProductionRatePerHour);
        }
        throw new NotImplementedException("Please implement the (static) AssemblyLine.ProductionRatePerHour() method");
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int)ProductionRatePerHour(speed)/60;

        throw new NotImplementedException("Please implement the (static) AssemblyLine.WorkingItemsPerMinute() method");
    }
}

File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        final int CARS = 221;

        switch (speed) {
            case 1:
            case 2:
            case 3:
            case 4:
                return speed * CARS;
            case 5:
            case 6:
            case 7:
            case 8:
                return speed * CARS * 0.9;
            case 9:
                return speed * CARS * 0.8;
            case 10:
                return speed * CARS * 0.77;
            default:
                return 0;
        }
        }

    /**
     * @param speed
     * @return
     */
    public int workingItemsPerMinute(int speed) {
        return (int)productionRatePerHour(speed) / 60;
    }
}
